app/controllers/api/posts_controller.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Api
+  class PostsController < ApiController
+    skip_before_action :require_login, only: %i[index post_responses]
+
+    def index
+      filter_params = query_params
+      posts = Post.posts_parents
+                  .order(filter_params[:order_by])
+                  .page(filter_params[:page])
+                  .per(filter_params[:page_size])
```

Comment:
Can we put the pagination logic in a single place? So we can use it here and elsewhere with ease